File: test/fixtures/sample.go
package main

import (
	"fmt"
	"math"
)

func add(a, b int) int {
	return a + b
}

type Point struct {
	x, y float64
}

func NewPoint(x, y float64) Point {
	return Point{x: x, y: y}
}

func (p Point) Distance(other Point) float64 {
	dx := p.x - other.x
	dy := p.y - other.y
	return math.Sqrt(dx*dx + dy*dy)
}

func main() {
	sum := add(1, 2)
	p1 := NewPoint(0.0, 0.0)
	p2 := NewPoint(3.0, 4.0)
	fmt.Printf("Sum: %d\n", sum)
	fmt.Printf("Distance: %f\n", p1.Distance(p2))
}

File: test/fixtures/sample.rs
fn add(a: i32, b: i32) -> i32 {
    a + b
}

struct Point {
    x: f64,
    y: f64,
}

impl Point {
    fn new(x: f64, y: f64) -> Self {
        Self { x, y }
    }

    fn distance(&self, other: &Point) -> f64 {
        let dx = self.x - other.x;
        let dy = self.y - other.y;
        (dx * dx + dy * dy).sqrt()
    }
}

fn main() {
    let sum = add(1, 2);
    let p1 = Point::new(0.0, 0.0);
    let p2 = Point::new(3.0, 4.0);
    println!("Sum: {}", sum);
    println!("Distance: {}", p1.distance(&p2));
}
